### DISCUSSION TITLE
Http Method Match in K8S Controller

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [dotnet/yarp#1825] [Ideas] [answered]
I have a proxied endpoint that requires authorization based on the Http Method. For `POST`, the endpoint accepts anonymous requests; for everything else it requires authorization. It's something akin to a user endpoint where `POST` represents signing up to an application.

I can easily codify this in configuration-based routes using the `route.Match.Methods` property,  plus _either_ a higher priority `Order` for the `POST` endpoint  _or_ spelling out the remaining HTTP methods on the other. However, this doesn't seem possible to do when using the Ingress controller as it does not have an http-methods annotation.

If such an annotation _were_ supported, this would require creating two `Ingress` resources with the same path and host. So far as I am aware, this is technically a valid thing to do per K8S. The Ingress Controller should be able to support such a case since it uses the `Ingress` name to construct the route's, so there'd be no conflict.

Is this something that could in theory be supported, or was it explicitly not considered? Or am I overlooking something that would prevent it?

**Comment by MihaZupan** [comment, marked as answer]
I don't see a reason it couldn't be supported as long as k8s is fine with duplicate (except for annotations) Ingress definitions.

Moved to issue #1828
